manager/controlapi/secret.go

```diff
@@ -157,12 +169,16 @@ func (s *Server) ListSecrets(ctx context.Context, request *api.ListSecretsReques
 //   or if the secret data is too long or contains invalid characters.
 // - Returns an error if the creation fails.
 func (s *Server) CreateSecret(ctx context.Context, request *api.CreateSecretRequest) (*api.CreateSecretResponse, error) {
+	err := s.populateSecretFromPlugin(ctx, request.Spec)
```

Comment:
Another alternative would be to put this resolution in the dispatcher, so the secret is fetched at the time the task is sent to the node where it will run. This is what I suggested earlier. The advantages would be:

- swarmkit would not be modifying the secret `Spec`. Specs are meant to be under the control of the user, and so far swarmkit never changes them. We're considering signing specs with a user-controlled key in the future.
- Calling `UpdateSecret` (for example, to change the secret's labels) would not have side effects on the payload, if it happened to change in the backend.
- We would avoid storing a copy of the secret payload inside the Raft datastore, which people may not want to do for security reasons.

The disadvantages would be:

- Inability to access the secrets backend could block deploying tasks, not just creating/updating secrets.
- To avoid redundant queries to the backend for each of N tasks that reference a secret, it would make sense to have an in-memory cache of secret payloads with a time-to-live, but this would add complexity.